packages/caver-klay/caver-klay-accounts/src/account/accountForUpdate.js

```diff
@@ -0,0 +1,69 @@
+const isValidRole = require('../../../../caver-utils').isValidRole
+
+class AccountForUpdate {
+	constructor(address, keyForUpdate, options) {
+		this.address = address
+		this.keyForUpdate = keyFormatter(keyForUpdate, options)
+	}
+
+	fillUpdateObject(updateObject) {
+		delete updateObject.key
+		Object.assign(updateObject, this.keyForUpdate)
+	}
+}
+
+function keyFormatter(keyForUpdate, options) {
+	const keyObject = {}
+
+	if (typeof keyForUpdate === 'string') {
+		if (options) throw new Error(`Failed to keyFormatter for AccountForUpdate: AccountKeyPublic/legacyKey/failKey cannot have options`)
+		switch(keyForUpdate) {
+			case 'legacyKey':
+				keyObject.legacyKey = true
+				break
+			case 'failKey':
+				keyObject.failKey = true
+				break
+			default:
+				keyObject.publicKey = keyForUpdate
```

Comment:
It seems that this keyForUpdate should be a public key. Don't we need to check its validity?

Reply:
If that is publicKey, parameter will be came with just string type. Do you mean check the validation of publicKey format?

Reply:
Yes. I think so. This will be done elsewhere?

Reply:
Validate function for public key is added :)